forge/src/main/java/me/shedaniel/architectury/registry/forge/TradeRegistryImpl.java

```diff
@@ -0,0 +1,65 @@
+package me.shedaniel.architectury.registry.forge;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMaps;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import me.shedaniel.architectury.forge.ArchitecturyForge;
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.minecraft.world.entity.npc.VillagerTrades;
+import net.minecraft.world.entity.npc.WanderingTrader;
+import net.minecraftforge.event.village.VillagerTradesEvent;
+import net.minecraftforge.event.village.WandererTradesEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.*;
+
+@Mod.EventBusSubscriber(modid = ArchitecturyForge.MOD_ID)
+public class TradeRegistryImpl {
+    
+    private static final Map<VillagerProfession, Int2ObjectMap<List<VillagerTrades.ItemListing>>> TRADES_TO_ADD = new HashMap<>();
+    private static final Int2ObjectMap<List<VillagerTrades.ItemListing>> WANDERER_TRADES_TO_ADD = new Int2ObjectOpenHashMap<>();
```

Comment:
`Boolean2ObjectMap` would make more sense here (as the key is pretty much the value of `rare`)

Reply:
A Boolean2ObjectMap does not exist within fastutil

Reply:
Ah, I'd just use two list fields then. The outer map isn't really needed.

Reply:
done